skills_ml/algorithms/aggregators/__init__.py

```diff
@@ -0,0 +1,60 @@
+from collections import Counter, defaultdict
+
+
+class JobAggregator(object):
+    def __init__(self):
+        self.initialize_counts()
+
+    def initialize_counts(self):
+        self.group_values = defaultdict(Counter)
+        self.rollup = defaultdict(Counter)
+
+    def accumulate(self, job_posting, job_key, groups):
+        """Incorporates the data from a single job posting
+
+        Args:
+            job_posting (dict) a job postings in common schema format
+            job_key (str) an element of the job posting (say, a title) that is
+                being used for aggregation
+            groups (iterable of tuples) the aggregatable groups (say, CBSAs)
+                that this job posting belongs to
+        """
+        value = self.value(job_posting)
+        for group_key in groups:
+            full_key = group_key + (job_key,)
```

Comment:
… and then define the `full_key = (group_key, job_key)`.  This way the aggregate group is always the first part of the key tuple.  This should make things a bit more flexible.